plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -886,11 +886,17 @@ func PodFitsHostPorts(pod *v1.Pod, meta interface{}, nodeInfo *schedulercache.No
 
 // search two arrays and return true if they have at least one common element; return false otherwise
 func haveSame(a1, a2 []string) bool {
-	for _, val1 := range a1 {
-		for _, val2 := range a2 {
-			if val1 == val2 {
-				return true
-			}
+	m := map[string]int{}
```

Comment:
This has a slightly different semantics in case of duplicates in slices.
We should check if it really works and document clearly the semantics about duplicates.